libs/muddle/src/rbc.cpp

```diff
@@ -703,7 +711,10 @@ bool RBC::CheckTag(RBCMessage const &msg)
  */
 bool RBC::SetPartyFlag(uint32_t sender_index, TagType tag, MessageType msg_type)
 {
-  assert(parties_.size() == current_cabinet_.size());
+  if (sender_index >= parties_.size())
+  {
+    return false;
+  }
```

Comment:
I don't think the assert should be replaced by something else here. If the assert fails this means there is a race condition between parties being reset and a message being processed and this should be caught and fixed. If we replace this with this if statement then we will not be alerted by this at all. I don't expect the assert to fail as as the sender index is computed within the lock in OnRBCMessage.